src/main/java/org/broadinstitute/hellbender/utils/haplotype/Haplotype.java

```diff
@@ -98,11 +98,21 @@ public Haplotype trim(final Locatable loc) {
             return null;
         }
 
-        // note: trimCigarByReference does not remove leading indels, while getBasesCoveringRefInterval does remove bases
-        // of leading indels.  We must remove leading indels from the Cigar manually.
+        // note: trimCigarByReference does not remove leading or trailing indels, while getBasesCoveringRefInterval does remove bases
+        // of leading and trailing insertions.  We must remove leading and trailing insertions from the Cigar manually.
+        // we keep leading and trailing deletions because these are necessary for haplotypes to maintain consistent reference coordinates
         final Cigar newCigar = AlignmentUtils.trimCigarByReference(getCigar(), newStart, newStop).getCigar();
-        final Cigar leadingIndelTrimmedNewCigar = newCigar.getFirstCigarElement().getOperator().consumesReferenceBases() ? newCigar :
-                new CigarBuilder().addAll(newCigar.getCigarElements().subList(1, newCigar.numCigarElements())).make();
+        final boolean leadingInsertion = !newCigar.getFirstCigarElement().getOperator().consumesReferenceBases();
```

Comment:
It would, but we never produce haplotypes with softclips (I think) and if we do we don't let them survive this far.  There may be a future refactoring to decouple the `Haplotype` from the assumption that it is necessarily produced by our particular combination of assembly, padding, Smith-Waterman, and trimming, but the time for that is not now.